src/helpers/shared/pagination.js

```diff
@@ -13,3 +13,22 @@ export const defaultCompactSettings = {
 export const getCurrentPage = (limit = 1, offset = 0) => Math.floor(offset / limit) + 1;
 
 export const getNewPage = (page = 1, offset) => (page - 1) * offset;
+
+export const getPaginationFromUrl = (defaultPagination, initialLoad) => {
+  const searchParams = new URLSearchParams(location.search);
+
+  isNaN(parseInt(searchParams.get('per_page'))) && searchParams.set('per_page', defaultPagination.limit);
+  const limit = parseInt(searchParams.get('per_page'));
+  isNaN(parseInt(searchParams.get('page'))) && searchParams.set('page', initialLoad ? 1 : defaultPagination.offset / limit + 1);
+  const offset = (parseInt(searchParams.get('page')) - 1) * limit;
+
+  window.history.replaceState({}, '', `${location.pathname}?${searchParams.toString()}`);
+  return { limit, offset };
+};
+
+export const setPaginationToUrl = (limit, offset = 0) => {
+  const searchParams = new URLSearchParams(location.search);
+  searchParams.set('per_page', limit);
```

Comment:
@mmenestr Is there a preferred name for these pagination variables? Throughout the platform I see `offset` paired with `limit` (patch and systems table), `page` paired with `page_size` (CVE table). 

I don't think I've seen `per_page` anywhere.

Reply:
@john-dupuy thanks for catching that, I think that using `page` number in the URL makes better sense to the user than `offset` value, I didn't realise that the limit value has different names throughout the platform so I'll rename it with any preferred name @mmenestr